source/jormungandr/jormungandr/autocomplete/elastic_search.py

```diff
@@ -204,3 +208,124 @@ def get(self, request, instance):
             return res['hits']
         except ConnectionError:
             raise TechnicalError("world wide autocompletion service not available")
+
+
+def create_admin_field(geocoding):
+        if not geocoding:
+            return None
+        admin_list = geocoding.get('admin', {})
+        response = []
+        for level, name in admin_list.iteritems():
+            response.append({
+                "insee": geocoding.get('TODO'),
+                "name": name,
+                "level": int(level.replace('level', '')),
+                "coord": {
+                    "lat": geocoding.get('TODO'),
+                    "lon": geocoding.get('TODO')
+                },
+                "label": name,
+                "id": geocoding.get('TODO'),
+                "zip_code": geocoding.get('TODO')
+            })
+        return response
+
+
+class AdminField(fields.Raw):
+    def output(self, key, obj):
+        if not obj:
+            return None
+        geocoding = obj.get('properties', {}).get('geocoding', {})
+        return create_admin_field(geocoding)
+
+
+class AddressField(fields.Raw):
+    def output(self, key, obj):
+        if not obj:
+            return None
+
+        coordinates = obj.get('geometry', {}).get('coordinates', [])
+        if len(coordinates) == 2:
+            lon = coordinates[0]
+            lat = coordinates[1]
+        else:
+            lon = None
+            lat = None
+
+        geocoding = obj.get('properties', {}).get('geocoding', {})
+
+        housenumber = geocoding.get('housenumber')
+        return {
+            "id": geocoding.get('id'),
+            "coord": {
+                "lon": lon,
+                "lat": lat,
+            },
+            "house_number": housenumber if ((housenumber is not None) & (housenumber != "")) else "0",
```

Comment:
I think you can rewrite this with:

`"house_number": geocoding.get('housenumber') or '0'`